### PR TITLE
Move obs factory request into bs request model

### PR DESCRIPTION
**Pull request by bgeuken**
* Drops ObsFactory::Request model and moves methods we need and use to bs request model.
  Some of the old methods were only needed to wire the BsRequest methods to  ObsFactory::Requests others were not used, like 'superseded_by'.
* Refactors ObsFactory::StagingProject#selected_requests
* Refactors ObsFactory::StagingProject#iso

### CODE REVIEW

File: src/api/app/models/obs_factory/staging_project.rb [hunk 1]
```diff
@@ -178,6 +169,7 @@ def missing_reviews
         attribs = [:by_group, :by_user, :by_project, :by_package]
 
         (open_requests + selected_requests).uniq.each do |req|
+          package = req.bs_request_actions.first.target_package
```
Comment (coolo) [inline src/api/app/models/obs_factory/staging_project.rb]: you have this now in 3 places. Once in this model and twice in views - not worth a model function ?

Comment (bgeuken) [reply src/api/app/models/obs_factory/staging_project.rb]: I thought about it, but what would be the name, `first_target_package`? :thinking: I guess we could do that.

Comment (coolo) [reply src/api/app/models/obs_factory/staging_project.rb]: you can also name it 'nick_name' - and possibly make it smarter over time :)

As maintenance requests would use a different nick name.

Comment (bgeuken) [reply src/api/app/models/obs_factory/staging_project.rb]: I think what I didn't like about it is that it feels a bit arbitrary to make the first request action special. But it seems to be a thing. So let's add a helper method.

File: src/api/app/presenters/obs_factory/staging_project_presenter.rb [hunk 2]
```diff
@@ -61,7 +61,7 @@ def classified_requests
         reviews[r[:request]] << r
       end
       requests.each do |req|
-        r = { number: req.number, package: req.package }
+        r = { number: req.number, package: req.bs_request_actions.first.target_package }
```
Comment (DavidKang) [inline src/api/app/presenters/obs_factory/staging_project_presenter.rb]: Could you use something more explainable than `r`?

Comment (bgeuken) [reply src/api/app/presenters/obs_factory/staging_project_presenter.rb]: I didn't introduce this so it's a bit out of scope for this PR. But the change is small enough. I am going to add another commit in a moment

**Comment by bgeuken** [comment]
@Ana06 Check the commit messages. The first one is squashing the two conditions by converting nil into an array. The second one simplifies setting `@selected_requests` by introducing a private method.

The other reason for the split was to make the changes more visible and avoid misunderstandings like [this](https://github.com/openSUSE/open-build-service/pull/5440/commits/64382f9903ca1baa37583390507c1136d1969f13#r204326875) :-)

**Comment by bgeuken** [comment]
Updated again per discussion with @Ana06

File: src/api/app/models/obs_factory/staging_project.rb [hunk 3]
```diff
@@ -321,5 +311,12 @@ def set_buildinfo
         @broken_packages = @broken_packages.reject { |p| p['state'] == 'unresolvable' }
       end
     end
+
+    private
+
+    def fetch_requests_from_meta
+      ids = meta["requests"].try(:map) { |i| i['id'] }
```
Comment (Ana06) [inline src/api/app/models/obs_factory/staging_project.rb]: is the `to_i` that was previous there not anymore needed?

**Pull request merged**